Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -184,34 +227,51 @@ void HiresTexture::Prefetch()
       }
       if (iter != s_textureCache.end())
       {
-        for (const Level& l : iter->second->m_levels)
+        for (const Level& level : iter->second->m_levels)
         {
-          size_sum += l.data_size;
+          prefetched_size_sum += level.data_size;
+          // a texture without mipmaps is considered to have 1 mip level
         }
       }
     }
 
-    if (s_textureCacheAbortLoading.IsSet())
+    if (s_textureCacheAbortPrefetching.IsSet())
     {
       return;
     }
 
-    if (size_sum > max_mem)
+    // Stops prefetching to avoid getting out of RAM completely, crash prone
+    // Currently no way to know how much RAM is used by OS and other applications
+    if (prefetched_size_sum >= delegated_max_mem)
     {
       Config::SetCurrent(Config::GFX_HIRES_TEXTURES, false);
 
+      WARN_LOG(VIDEO,
+               "Prefetcher: Warning, custom textures prefetching aborted after %.3f MB."
+               " Maximum delegated memory limit of %.3f breached.",
+               prefetched_size_sum / BYTES_IN_MEGABYTES, delegated_max_mem / BYTES_IN_MEGABYTES);
       OSD::AddMessage(
           StringFromFormat(
               "Custom Textures prefetching after %.1f MB aborted, not enough RAM available",
-              size_sum / (1024.0 * 1024.0)),
-          10000);
+              prefetched_size_sum / BYTES_IN_MEGABYTES),
+          20000);
       return;
     }
-  }
-  u32 stoptime = Common::Timer::GetTimeMs();
-  OSD::AddMessage(StringFromFormat("Custom Textures loaded, %.1f MB in %.1f s",
-                                   size_sum / (1024.0 * 1024.0), (stoptime - starttime) / 1000.0),
-                  10000);
+
+  }  // End Prefetching
+  prefetch_stop_time_total = Common::Timer::GetTimeMs();
+
+  const double pref_sum_total_mb = prefetched_size_sum / BYTES_IN_MEGABYTES;
+  const double pref_time_total_sec =
+      (prefetch_stop_time_total - prefetch_start_time_total) / MILISEC_IN_SECONDS;
+
+  std::string pref_final_msg =
+      StringFromFormat("Prefetcher: Total of %u custom textures loaded, %.1f MB in %.1f seconds.",
+                       counter_textures_mips_amount, pref_sum_total_mb, pref_time_total_sec);
+
+  OSD::AddMessage(pref_final_msg, 20000);
+
+  INFO_LOG(VIDEO, pref_final_msg.c_str());
```

Comment:
Replace `pref_final_msg.c_str()` with `%s, pref_final_msg.c_str()`.

Reply:
None of the possible combinations I tried works.

Reply:
Sorry, my bad. It's supposed to be `"%s"` where I said `%s`